### ISSUE TITLE
GitHub Actions Failed Jobs - Deprecated Go version Unix SDK Tests

### ISSUE THREAD

**Issue by syall** [aws/aws-sdk-go-v2#1946]
### Describe the bug

Looks like since around 11/15/2022, Deprecated Go version Unix SDK Tests Github Action Jobs have failed for the following tests in `internal/sync/singleflight/singleflight_test.go`:

- `TestPanicDoChan`
- `TestPanicDoSharedByDoChan`

Scope affects normal commit GitHub actions and PR checks.

### Expected Behavior

Passing tests.

### Current Behavior

Example Logs:

```log
--- FAIL: TestPanicDoChan (0.01s)
    singleflight_test.go:256: /var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/go-build327660369/b224/singleflight.test -test.run=TestPanicDoChan -test.v:
    singleflight_test.go:264: Test subprocess failed, but the crash isn't caused by panicking in DoChan
--- FAIL: TestPanicDoSharedByDoChan (0.02s)
    singleflight_test.go:310: /var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/go-build327660369/b224/singleflight.test -test.run=TestPanicDoSharedByDoChan -test.v:
    singleflight_test.go:318: Test subprocess failed, but the crash isn't caused by panicking in Do
```

---

Latest commit on date of opening issue:

Github Actions Job Log from 2022-12-01: https://github.com/aws/aws-sdk-go-v2/actions/runs/3595829931/jobs/6056686631#step:4:321

---

First observed failure:

Github Actions Job Log from 2022-11-15: https://github.com/aws/aws-sdk-go-v2/actions/runs/3473512441/jobs/5806469472#step:4:314

### Reproduction Steps

See Current Behavior for Github Actions Job logs.

### Possible Solution

N/A.

### Additional Information/Context

N/A.

### AWS Go SDK V2 Module Versions Used

N/A.

### Compiler and Version used

N/A.

### Operating System and version

N/A.

**Comment by RanVaknin** [comment]
Hi @syall ,

We are working on a fix to exclude the Go versions that are unsupported on `mac os latest`

Hopefully this will solve the issue.

Thanks,
Ran~

**Comment by syall** [comment]
Fixed with aws/aws-sdk-go-v2#1971
